lib/mix/lib/mix/tasks/test.ex

```diff
@@ -412,8 +412,15 @@ defmodule Mix.Tasks.Test do
             Mix.shell().info("There are no tests to run")
 
           true ->
-            message = "Paths given to \"mix test\" did not match any directory/file: "
-            raise_or_error_at_exit(message <> Enum.join(files, ", "), opts)
+            message =
+              "Paths given to \"mix test\" did not match any directory/file: " <>
+                Enum.join(files, ", ")
+
+            if Mix.ProjectStack.recursing() do
```

Comment:
This is private API and I don't think we should be accessing it here - our tasks should not "cheat" just because they are also part of Mix. We would first need to expose it at `Mix.Project`. :)